lib/repository.js

```diff
@@ -622,6 +622,74 @@ Repository.prototype.createCommitBuffer = function(
     });
 };
 
+/**
+ * Create a commit that is digitally signed
+ *
+ * @async
+ * @param {Signature} author
+ * @param {Signature} committer
+ * @param {String} message
+ * @param {Tree|Oid|String} Tree
+ * @param {Array} parents
+ * @param {String} signature_field typically "gpgsig"
+ * @param {Function} onSignature Callback to be called with string to be signed
+ * @return {Oid} The oid of the commit
+ */
+Repository.prototype.createCommitWithSignature = function(
+    author,
+    committer,
+    message,
+    tree,
+    parents,
+    signature_field,
+    onSignature,
+    callback) {
```

Comment:
I don't think we need a callback param here because we're using a promise already, can we remove this param?

Reply:
:) I'm happy to remove it if it's not needed.

It's referenced on line 690. Is it not needed for the rejection call?
I see other places in this file with a similar pattern. like `Repository.getReferences`

Reply:
I think those are old patterns before the library was heavily promisified. It's more of a remnant of old code style and a smell in new code.

Yes, please remove the ref to callback from 690 and let's get this merged!

Reply:
sounds good. I pushed a commit removing the callback